src/nl/rubensten/texifyidea/window/PreviewForm.kt

```diff
@@ -0,0 +1,126 @@
+package nl.rubensten.texifyidea.window
+
+import com.google.common.io.Files
+import java.io.*
+import java.util.concurrent.TimeUnit
+import javax.imageio.ImageIO
+import javax.swing.*
+
+class PreviewForm {
+    var panel: JPanel? = null
+    private var equation_area: JTextArea? = null
+    private var output_area: JTextArea? = null
+    private var output_panel: JPanel? = null
+    private var equation_panel9: JPanel? = null
+    private var preview_panel: ImagePanel? = null
+
+    private val preamble = """
+    \usepackage{amsmath,amsthm,amssymb,amsfonts}
+    \usepackage{color}
+    """.trimIndent()
+
+    fun set_equation_text(equation_text: String) {
+        equation_area!!.text = equation_text
+
+        try {
+            val tmpdir = Files.createTempDir()
```

Comment:
IntelliJ says that `Files` is 'unstable'... Maybe find an alternative... otherwise, just suppress the warning with `@Suppress("UnstableApiUsage")`

Also rename to `tempDirectory` for better readability.